src/p2p/swarm.rs

```diff
@@ -101,40 +121,75 @@ impl NetworkBehaviour for SwarmApi {
     fn addresses_of_peer(&mut self, peer_id: &PeerId) -> Vec<Multiaddr> {
         log::trace!("addresses_of_peer {}", peer_id);
         if let Some(addr) = self.connected_peers.get(peer_id).cloned() {
```

Comment:
this one could be `self.connected_peers.get(peer_id).cloned().unwrap_or_default()` instead of an `if let`

Reply:
Another good catch, thanks!